tests/rbr/AdjustRegionsTest.java

```diff
@@ -0,0 +1,82 @@
+    Set<Character> op = new HashSet<>();
+    private Region original;
+    private RBR rbr;
+
+    @Before
+    public void setUp() {
+        Graph graph = RegularGraphBuilder.generateGraph(4,4);
+        rbr = new RBR(graph);
+        Set<Vertex> dst = new HashSet<>();
+        ip.add('I');
+        op.add('N');
+        dst.add(new Vertex("0.1"));
+        dst.add(new Vertex("0.2"));
+        dst.add(new Vertex("0.3"));
+        dst.add(new Vertex("1.2"));
+        dst.add(new Vertex("1.3"));
+        dst.add(new Vertex("2.2"));
+        dst.add(new Vertex("2.3"));
+        dst.add(new Vertex("3.3"));
+        original = new Region(ip, dst, op);
+    }
+
+    @Test
+    public void regionWithoutOutsidersIsntAdjusted() throws Exception {
+        Set<Vertex> dstRegionWithoutOutsiders = new HashSet<>();
+        dstRegionWithoutOutsiders.add(new Vertex("1.1"));
+        Region regionWithoutOutsiders = new Region(ip, dstRegionWithoutOutsiders, op);
+        ArrayList<Region> adjustedRegionWithoutOutsiders = rbr.adjustedRegionsFrom(regionWithoutOutsiders);
+        ArrayList<Region> arrayRegionWithoutOutsiders = new ArrayList<>();
+        arrayRegionWithoutOutsiders.add(regionWithoutOutsiders);
+        Assert.assertEquals(adjustedRegionWithoutOutsiders, arrayRegionWithoutOutsiders);
+    }
+
+    @Test
+    public void regionWithoutDestinationsIsntAdjusted() throws Exception {
+        Set<Vertex> dstRegionWithoutDestinations = new HashSet<>();
+        Region regionWithoutDestinations = new Region(ip, dstRegionWithoutDestinations, op);
+        ArrayList<Region> adjustedRegionWithoutDestinations = rbr.adjustedRegionsFrom(regionWithoutDestinations);
+        ArrayList<Region> arrayRegionWithoutDestinations = new ArrayList<>();
+        arrayRegionWithoutDestinations.add(regionWithoutDestinations);
+        Assert.assertEquals(adjustedRegionWithoutDestinations, arrayRegionWithoutDestinations);
+    }
+
+    @Test
+    public void adjustedRegionsHaveSameInputPortsAsOriginal() throws Exception {
+        for(Region adjustedRegion : rbr.adjustedRegionsFrom(original)) {
+            Assert.assertEquals(adjustedRegion.inputPorts(), original.inputPorts());
+        }
+    }
+
+    @Test
+    public void adjustedRegionsHaveSameOutputPortsAsOriginal() throws Exception {
+        for(Region adjustedRegion : rbr.adjustedRegionsFrom(original)) {
+            Assert.assertEquals(adjustedRegion.outputPorts(), original.outputPorts());
+        }
+    }
+
+    @Test
+    public void adjustedRegionsHaveSameDestinationsAsOriginal() throws Exception {
+        Set<Vertex> destinationsAdjustedRegions = new HashSet<>();
+        for(Region adjustedRegion : rbr.adjustedRegionsFrom(original)) {
+            destinationsAdjustedRegions.addAll(adjustedRegion.destinations());
+        }
+        Assert.assertEquals(destinationsAdjustedRegions, original.destinations());
+    }
+
+    @Test
+    public void adjustedRegionsWithoutOutsiders() throws Exception {
```

Comment:
Don't you think that 'adjustedRegionsHaveNoOutsider' would reveal more the intent of this test?